teuthology/test/test_repo_utils.py

```diff
@@ -12,13 +13,14 @@
 
 
 class TestRepoUtils(object):
-    src_path = '/tmp/empty_src'
+    temp_path = tempfile.mkdtemp(prefix='test_repo', dir='/tmp')
```

Comment:
nit, i think it'd be better to allow user to customize the default directory by specifying environmental variables like `TMPDIR`, `TEMP` or `TMP`. and `/tmp` is the first directory used by `tempfile.mkdtemp()` tries to create a tempfile on UNIX-like OSes.

Reply:
also, `temp_path` is created when this `class` is evaluated. but you `rm -rf` it whenever a test method is executed.

if this directory is supposed to be shared among the test methods, we should create it in `setup_class()` and nuke it in `teardown_class()`. and, yeah, initialize `dest_path` in `setup_method()`.

Reply:
> nit, i think it'd be better to allow user to customize the default directory by specifying environmental variables like `TMPDIR`, `TEMP` or `TMP`. and `/tmp` is the first directory used by `tempfile.mkdtemp()` tries to create a tempfile on UNIX-like OSes.
This change is for resolving test failures in multiuser environment. I don't understand what exactly you want, which of the variable above you want to use? Do you want to use 'dir=sys.getenv('TEMP', '/tmp')`? Could you provide code example for it.

Reply:
> also, `temp_path` is created when this `class` is evaluated. but you `rm -rf` it whenever a test method is executed.
> 
So, you mean that *teardown_method* should be renamed to `teardown_class`?

> if this directory is supposed to be shared among the test methods, we should create it in `setup_class()` and nuke it in `teardown_class()`. and, yeah, initialize `dest_path` in `setup_method()`.

We should create a temporary directory in the same place where we setup all those path variables, they are in constructor, do you like to move them to setup_class method?

Reply:
@kshtsk see https://github.com/enthought/Python-2.7.3/blob/master/Lib/tempfile.py#L154 . i'd suggest remove `dir` parameter.

Reply:
> So, you mean that *teardown_method* should be renamed to `teardown_class`?

no. `self.dest_path` is created by tests, for instance, see https://github.com/ceph/teuthology/pull/1257/files#diff-3bee1a3b7d5220a2093743ab3f0004f9R127 . so we should at least remove that file in `teardown_method()`. but i think the better approach is probably to remove the whole directory in `teardown_method()`. and create / initialize `temp_path` in `setup_method()`, unless we want to reuse the parent directory. but i cannot think of a reason to do so.

Reply:
> @kshtsk see https://github.com/enthought/Python-2.7.3/blob/master/Lib/tempfile.py#L154 . i'd suggest remove `dir` parameter.

Agree.

Reply:
@tchaikov Changes pushed, could you please review again.